### PR TITLE
Matrix Subtraction in C

### LINKED ISSUE #5580: Matrix Subtraction in C
## 🚀 Feature

(A clear and concise description of what the feature is.)

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## Pitch

Matrix Subtraction in C

### PR DESCRIPTION
**Pull request by abhijeet141**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Matrix Subtraction in C

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #5580

### CODE REVIEW

**Comment by abhijeet141** [comment]
Respected Mentors please review my PR

File: C/math/matrix_subtraction.c [hunk 1]
```diff
@@ -0,0 +1,78 @@
+//C program to implement Matrix Subtraction
```
Comment (HarshCasper) [inline C/math/matrix_subtraction.c]: Add a bit more information about your code here

Comment (RaiyanMahin) [inline C/math/matrix_subtraction.c]: add a new line after `#include<stdio.h>`

File: C/math/matrix_subtraction.c [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+#include <stdio.h>
+void sub(int r1, int c1, int mat1[r1][c1], int mat2[r1][c1])
```
Comment (RaiyanMahin) [inline C/math/matrix_subtraction.c]: add one line description about the function.

File: C/math/matrix_subtraction.c [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    printf("\nEnter the number of rows in matrix 1 = ");
+    scanf("%d", &r1);
+    printf("\nEnter the number of columns in matrix 1 = ");
+    scanf("%d", &c1);
+    printf("\nEnter the number of rows in matrix 2 = ");
+    scanf("%d", &r2);
+    printf("\nEnter the number of columns in matrix 2 = ");
+    scanf("%d", &c2);
+    int mat1[r1][c1], mat2[r2][c2];
+    //If number of rows and columns of matrix 1 and matrix 2 are different then they cannot be subtracted
```
Comment (RaiyanMahin) [inline C/math/matrix_subtraction.c:44]: use multi line comment

File: C/math/matrix_subtraction.c [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+Enter the values in matrix 2 -->
+2 3
+4 5
+
+
+Matrix after Subtraction is -->
+2       2
+2       2
+*/
```
Comment (RaiyanMahin) [inline C/math/matrix_subtraction.c]: add new line at EOF

File: C/math/matrix_subtraction.c [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+void sub(int r1, int c1, int mat1[r1][c1], int mat2[r1][c1])
+{
+    int mat3[r1][c1];
+    for(int i = 0; i < r1; i++)
+        for(int j = 0; j < c1; j++)
+        //Subtracting both the matrix
+            mat3[i][j] = mat1[i][j] - mat2[i][j];
+             printf("Matrix after subtraction is -->\n");
+    for(int i = 0; i < r1; i++)
+    {   for(int j = 0; j < c1; j++)
```
Comment (RaiyanMahin) [inline C/math/matrix_subtraction.c]: align code properly

**Review by RaiyanMahin — CHANGES_REQUESTED** [review]

File: C/math/matrix_subtraction.c (hunk 4 above)
Comment (HarshCasper) [inline C/math/matrix_subtraction.c]: Add a new line at the EOF

File: C/math/matrix_subtraction.c (hunk 2 above)
Comment (HarshCasper) [inline C/math/matrix_subtraction.c]: Use meaningful variable names

File: C/math/matrix_subtraction.c [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+            scanf("%d", &mat1[i][j]);
+    printf("\n\nEnter the values in matrix 2 -->\n");
+    for(int i = 0; i < r1; i++)
+        for(int j = 0; j < c1; j++)
+            scanf("%d", &mat2[i][j]);
+    printf("\n\n");
+}
+int main()
+{
+    int i, j , r1, r2, c1, c2;
```
Comment (HarshCasper) [inline C/math/matrix_subtraction.c]: Use meaningful variable names

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by abhijeet141** [comment]
@HarshCasper I have made all changes please review it once

File: C/math/matrix_subtraction.c [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    scanf("%d", &column_2);
+    int matrix_1[row_1][column_1], matrix_2[row_2][column_2];
+    //If number of rows and columns of matrix 1 and matrix 2 are different then they cannot be subtracted
+            if (row_1 == row_2 && column_1 == column_2)
+            {
+                input(row_1, column_1, matrix_1, matrix_2);
+                subtract(row_1, column_1, matrix_1, matrix_2);
+            }
+            else
+                printf("Subtraction of the given matrices is not possible.");
```
Comment (ankitaggarwal23) [inline C/math/matrix_subtraction.c]: Remove the redundant comments and left align it one tab space is enough

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by abhijeet141** [comment]
@ankitaggarwal23 made changes please check

File: C/math/matrix_subtraction.c [hunk 8]
```diff
@@ -0,0 +1,82 @@
+/*
+Matrix is a 2-D Array.
+The Subtraction of a matrix is possible by subtracting the element of another matrix having same order.
+This code gives the output as as difference between two input matrices.
+*/
+#include <stdio.h>
```
Comment (ankitaggarwal23) [inline C/math/matrix_subtraction.c:6]: Add a new line after header

File: C/math/matrix_subtraction.c [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+            printf("Matrix after subtraction is -->\n");
+    for(int i = 0; i < row_1; i++)
+    {   for(int j = 0; j < column_1; j++)
+        {
+            printf("%d\t", matrix_3[i][j]);
+        }
+        printf("\n");
+    }
+}
+//Function that takes input of both matrices
```
Comment (ankitaggarwal23) [inline C/math/matrix_subtraction.c:26]: Add a new line above comments

File: C/math/matrix_subtraction.c [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+    for(int i = 0; i < row_1; i++)
+        for(int j = 0; j < column_1; j++)
+            scanf("%d", &matrix_1[i][j]);
+            printf("\n\nEnter the values in matrix 2 -->\n");
+    for(int i = 0; i < row_1; i++)
+        for(int j = 0; j < column_1; j++)
+            scanf("%d", &matrix_2[i][j]);
+            printf("\n\n");
+}
+int main()
```
Comment (ankitaggarwal23) [inline C/math/matrix_subtraction.c:40]: Add a new line above main

File: C/math/matrix_subtraction.c [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+  printf("\nEnter the number of rows in matrix 2 = ");
+  scanf("%d", &row_2);
+  printf("\nEnter the number of columns in matrix 2 = ");
+  scanf("%d", &column_2);
+  int matrix_1[row_1][column_1], matrix_2[row_2][column_2];
+  if (row_1 == row_2 && column_1 == column_2)
+{
+  input(row_1, column_1, matrix_1, matrix_2);
+  subtract(row_1, column_1, matrix_1, matrix_2);
+}
```
Comment (ankitaggarwal23) [inline C/math/matrix_subtraction.c]: Align it properly

File: C/math/matrix_subtraction.c [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+  else
+  printf("Subtraction of the given matrices is not possible.");
+}
+
+/*
+Time Complexity: O(r * c), where 'r' is the number of rows and 'c' is the number of columns
+Space Complexity: O(r * c)
```
Comment (ankitaggarwal23) [inline C/math/matrix_subtraction.c:63]: Add a new line after SC

File: C/math/matrix_subtraction.c [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+Enter the number of columns in matrix 1 = 2
+
+Enter the number of rows in matrix 2 = 2
+
+Enter the number of columns in matrix 2 = 2
+Enter the values in matrix 1 -->
+4 5
+6 7
+
+
```
Comment (ankitaggarwal23) [inline C/math/matrix_subtraction.c]: Remove one extra line

File: C/math/matrix_subtraction.c [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+Enter the values in matrix 2 -->
+2 3
+4 5
+
+
```
Comment (ankitaggarwal23) [inline C/math/matrix_subtraction.c]: Remove one extra line

File: C/math/matrix_subtraction.c [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+Matrix after Subtraction is -->
+2       2
+2       2
+*/
```
Comment (ankitaggarwal23) [inline C/math/matrix_subtraction.c:84]: Add a new line at the end

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by abhijeet141** [comment]
@ankitaggarwal23 please check

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**